libs/ngx-dsfr/badge/src/lib/testing/badge.component.spec.ts

```
@@ -0,0 +1,120 @@
+ */
+import { DebugElement } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+
+/**
+ * 3rd-party imports
+ */
+import { ThemeColor } from '@betagouv/ngx-dsfr';
+
+/**
+ * Internal imports
+ */
+import {
+  DsfrBadgeComponent,
+  EMPTY_LABEL_ERROR,
+  EMPTY_THEME_ERROR,
+  EMPTY_LABEL_AND_THEME_ERROR
+} from '../badge.component';
+import { TestHostComponent } from './test-host.component';
+import { DsfrBadgeHarness } from './badge.harness';
+
+
+describe('DsfrBadgeComponent', () => {
+  let fixture: ComponentFixture<TestHostComponent>;
+  let testHost: TestHostComponent;
+  let componentUnderTest: DsfrBadgeComponent;
+  let harnessLoader: HarnessLoader;
+  let dsfrBadgeHarness: DsfrBadgeHarness;
+
+  const testLabel: string = 'Test Label';
+  const testTheme: ThemeColor = ThemeColor.WARNING;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        DsfrBadgeComponent,
+        TestHostComponent
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    componentUnderTest = fixture.debugElement.query(
+      By.directive(DsfrBadgeComponent)
+    ).componentInstance;
+
+    /*
+     * We're retrieving a HarnessLoader HERE to be able to get
+     * the Test Harness for the DsfrBadgeComponent LATER.
+     *
+     * We're NOT doing both HERE since getHarness() triggers
+     * indirectly the Change Detection mechanism and thus,
+     * Data Binding and ngOnInit()...
+     */
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should be created', () => {
+    expect(componentUnderTest).toBeTruthy();
+  });
+
+  it('should throw an error when no label and no theme is provided', async () => {
```

`it('should throw an error when no label and no theme are provided'`